src/control-plane/backend/lambda/api/router/segment.ts

```diff
@@ -33,7 +33,7 @@ const segmentServ = new SegmentServ();
 router_segment.post(
   '',
   validate([
-    body().custom(isValidEmpty).custom(isXSSRequest),
+    body().custom(isValidEmpty),
```

Comment:
Is it safe to remove the XSS check?

Reply:
This API body contains metadata of events, and the **description** information may have HTML tags. The current logic has to remove this XSS check rule, otherwise the user group cannot be created. Or modify the current logic of passing parameters, can we skip the description information @jingnanl .

```
{
          "name": "_link_url",
          "dataType": "string",
          "category": "event",
          "displayName": {
            "en-US": "Link URL",
            "zh-CN": "外链地址"
          },
          "description": {
            "en-US": "The content of href in tag <a>",
            "zh-CN": "标签 <a>中href里的内容"
          },
          "eventName": "_click"
        }
```

Reply:
I do not see why we should include metadata of events when creating a segment.

we need to polish API implementation.

Reply:
It is because we save all the metadata used for rendering UI components in the backend. We need to change the frontend implementation instead of API to avoid that.

Reply:
Fix the issue on other PR